src/Passes/Source/Passes/ExpandStaticAllocation/ExpandStaticAllocation.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Passes/ExpandStaticAllocation/ExpandStaticAllocation.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <fstream>
+#include <iostream>
+
+namespace microsoft
+{
+namespace quantum
+{
+    llvm::PreservedAnalyses ExpandStaticAllocationPass::run(
```

Comment:
A doc comment would be helpful, especially for larger functions that are intended to be more widely used.